src/vcpkg/base/downloads.cpp

```diff
@@ -786,6 +786,10 @@ namespace vcpkg
             }
         }
 #endif
+        // Create directory in advance, otherwise curl will create it in 750 mode on unix style file systems.
+        const auto dir = download_path_part_path.parent_path();
```

Comment:
I'm not sure this is the right thing, there are a lot of edge cases where just chopping off the last path element will do the wrong thing. Consider something like `a/b/../out.exe`.